test/src/unit-backwards_compat.cc

```diff
@@ -443,3 +443,16 @@ TEST_CASE(
     }
   }
 }
+
+TEST_CASE(
+    "Backwards compatibility: Write to an array of older version",
+    "[backwards-compat][write-to-older-version]") {
+  std::string old_array_name(arrays_dir + "/non_split_coords_v1_4_0");
+  Context ctx;
+  try {
+    Array old_array(ctx, old_array_name, TILEDB_WRITE);
```

Comment:
FYI you can probably shorten this check using `REQUIRE_THROWS_AS`, e.g. https://github.com/TileDB-Inc/TileDB/blob/dev/test/src/unit-cppapi-array.cc#L553